be/src/storage/tablet_updates.cpp

```diff
@@ -1097,6 +1100,9 @@ Status TabletUpdates::_wait_for_version(const EditVersion& version, int64_t time
     if (!(_edit_version_infos[_apply_version_idx]->version < version)) {
         return Status::OK();
     }
+    if (timeout_ms == 0) {
```

Comment:
When `timeout_ms == 0`, then  `_wait_for_version` just return OK may break the semantics of `_wait_for_version`. If someone who want to check version and not need to wait, he will just get OK by mistake. 

You can do this check outside `_wait_for_version`, eg:
```
if (wait_time > 0) {
   st = _wait_for_version(EditVersion(version, 0), wait_time, ul);
}
```

Reply:
> When `timeout_ms == 0`, then `_wait_for_version` just return OK may break the semantics of `_wait_for_version`. If someone who want to check version and not need to wait, he will just get OK by mistake.
> 
> You can do this check outside `_wait_for_version`, eg:
> 
> ```
> if (wait_time > 0) {
>    st = _wait_for_version(EditVersion(version, 0), wait_time, ul);
> }
> ```

Great advice, changed in https://github.com/StarRocks/starrocks/pull/17101/commits/408f82035b6a4476d50d0f24dd37fdbeb6a3dcf3